src/pages/auth/signin.tsx

```diff
@@ -0,0 +1,48 @@
+import { getProviders, signIn } from 'next-auth/react'
+import { useRouter } from 'next/router'
+
+import type { Provider } from 'next-auth/providers'
+
+// https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/pages/signin.tsx
+
+/**
+ * Custom sign in page.
+ * TODO: needs to style it :)
+ */
+
+type Props = {
+  providers: Provider
+}
+
+const SignIn = ({ providers }: Props) => {
+  const { query } = useRouter()
+
+  return (
+    <>
+      {Object.values(providers).map((provider) => (
+        <div key={provider.name}>
+          <button
+            onClick={() =>
+              signIn(provider.id, {
+                callbackUrl: query?.redirect as string,
+              })
+            }
+            type="button"
+          >
+            Sign in with {provider.name}
+          </button>
+        </div>
+      ))}
+    </>
+  )
+}
+
+export default SignIn
```

Comment:
Does this page replace the `/user/auth` page?

Reply:
This should not be part of the git push, it can be deleted

Reply:
yeah, though we can change it to `user/auth` if we don't like the endpoint

Reply:
Nah it's fine, I just have to move nav logic to this page

Reply:
Wait I think you might be misunderstanding something. `/auth/signin` is the custom sign in page for next-auth. In the example we currently have, `/user/auth` is like a dashboard, where they click the "sign in" button and gets redirected to the default next-auth sign in page. 

`/auth/signin` is a replacement to next-auth's sign in page. TBH `/user/auth` shouldn't be really named "auth" since it only checks auth and doesn't authenticate (so more like a protected page). 

If you take a look at the flowchart in Gdrive, the user clicks on the "sign in" button on the home page to be redirected (using the useRedirect() hook) to `/auth/signin` and after signing in, they're automatically redirected to `/dashboard` 

For redirecting to different pages based on application status, I'd recommend using a callback for sign in
https://next-auth.js.org/configuration/callbacks#sign-in-callback 
Or if this doesn't work, I'd recommend moving the redirect logic to `/dashboard`

Reply:
shoot yeah you're right. The page I'm talking about is the redirect page isn't it.

Reply:
yep! 👌